packages/core/src/schema/node.ts

```diff
@@ -496,7 +496,7 @@ export function buildNodeFromSplits(
     ) {
       const textDoc = doc as TextNode;
       const node = new TextNode({
-        id_: idGenerator(i, textDoc),
+        id_: textDoc.id_ ?? idGenerator(i, textDoc),
```

Comment:
LGTM, could you also update other node creation parts?

Reply:
The issue is in buildNodeFromSplits as far as I can see, I have added imageNode id also